fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java

```diff
@@ -83,6 +90,19 @@ public void removeQueryMetadata() {
         }
     }
 
+    public List<String> listWhNames() {
+        Optional<ConnectorMetadata> connectorMetadata = getWarehouseMetaData();
```

Comment:
Warehouse has nothing to do with connector api, actually i think the best metadata object to refer to is `Catalog`, you can refer to `handleShowCatalogs()` to see how to implement `show warehouses`;